pkg/storage/gcp/bucket_mock.go

```diff
@@ -0,0 +1,95 @@
+package gcp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+type bucketMock struct {
+	db   map[string][]byte
+	lock sync.RWMutex
+}
+
+func newBucketMock() Bucket {
+	b := &bucketMock{}
+	b.db = make(map[string][]byte)
+	return b
+}
+
+type bucketReader struct {
+	io.Reader
+	*bucketMock
+}
+
+type bucketWriter struct {
+	*bucketMock
+	path string
+}
+
+func (r *bucketReader) Close() error {
+	r.bucketMock.lock.RUnlock()
+	return nil
+}
+
+func (r *bucketWriter) Close() error {
+	r.bucketMock.lock.Unlock()
+	return nil
+}
+
+func (r *bucketWriter) Write(p []byte) (int, error) {
```

Comment:
should this do a 

```go
r.lock.Lock()
defer r.lock.Unlock()
```

?

Reply:
Nope, as per the comment above. The lock is taken while returning the Writer, and released in Close